### ISSUE TITLE
JournalD logging broken in 0.22

### ISSUE THREAD

**Issue by northwestnodes-eric** [vectordotdev/vector#12966]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Problem

Using a standard Vector setup of a `journald` type source, transform and `datadog_logs` sink, it seems 0.22 stopped parsing/sending (can't really make that part out) log entries over to DataDog. Removing the transform portion of the pipeline does nothing.

As I just downgraded from 0.22 to 0.21.2, old log entries from a few hours ago got processed and sent over. As well as newer entries.

### Configuration

```text
We'd rather not share our vector.toml config but it has been sent to jszwedko on Discord.
```

### Version

0.22

### Example Data

N/A

### Additional Context

We run Ubuntu 20.04 LTS x64 on a virtual machine in Microsoft Azure. We do not believe this to be of any concern as downgrading vector to 0.21.2 fixed the problem described above.

**Comment by bruceg** [comment]
I played around with this a bit, and I have definitely been able to reproduce this in 0.22.0. However, I was not able to reproduce in our current master branch, so I think the solution to this is to forward-port the changes to the journald source from there (which have just been clean-ups so far) to the 0.22 branch. Namely, that would be:
* #12885 
* #12889 
* #12908

**Comment by jszwedko** [comment]
Just noting for reference that I bisected down the regression to 06aca83182ae6129000023b0709e217a454e0d0b

**Comment by jszwedko** [comment]
Looks like https://github.com/vectordotdev/vector/pull/12885 is the fix (ran `git bisect` again). We could think about just pulling in that one if the others introduce more risk. Curious what you think @bruceg . Given this is a patch release, we generally want to keep the risk of any additional regressions to a minimum.

**Comment by bruceg** [comment]
That would be the minimal fix, I'm sure. I would also want to include #12889, as it too is a bug fix albeit not a regression fix. #12908 is more of a testing enhancement, so isn't quite so stable-worthy.

**Comment by KannarFr** [comment]
(Will backport patch for Exherbo packaging if not).

**Comment by jszwedko** [comment]
Closing since this is fixed by https://github.com/vectordotdev/vector/pull/12885 . We'll be releasing it tomorrow.

**Comment by miton18** [comment]
> > Hi, can we expect a 0.22.1 with these fixes soon?
> 
> We're looking to release tomorrow or Thursday, we've got one more issue we would like to include in the patch.

Do you know when you will be able to do a release ?

**Comment by mr-karan** [comment]
Is this fixed in 0.22.2?

**Comment by jszwedko** [comment]
Hey @mr-karan ! Yes, this is fixed in 0.22.2. Let us know if you have any issues!
